Resource.watch failed when type is error and object is v1.Status

<h2>Pre-bug-report checklist</h2>

**1. This bug can be reproduced using kubectl**
- [ ] Yes [👉 Please report a bug to the Kubernetes GitHub 👈](https://github.com/kubernetes/kubernetes/issues/new/choose)
- [x] No

_If yes, it is more likely to be an Kubernetes bug unrelated to Cloudcoil. Please double check before submitting an issue to Cloudcoil._

**2. I have searched for [existing issues](https://github.com/cloudcoil/cloudcoil/issues)**
- [x] Yes

**3. This bug occurs in Cloudcoil when...**
- [x] Performing CRUD operations on Resources
- [ ] Using the Cloudcoil code-generator
- [ ] Parsing/Exporting from files or YAML
- [ ] other: 

<h2>Bug report</h2>

**Describe the bug**
_A clear and concise description of what the bug is:_

When calling `some_resource.watch(...)`, incoming event is error, and obj type is v1.Status, cloudcoil fails to proceed.

_Error log if applicable_:
```
>                       obj = self.kind.model_validate(event["object"])
E                       pydantic_core._pydantic_core.ValidationError: 2 validation errors for Dynamic<ResourceName>
E                       apiVersion
E                         Input should be '<API Group And Version>' [type=literal_error, input_value='v1', input_type=str]
E                           For further information visit https://errors.pydantic.dev/2.10/v/literal_error
E                       kind
E                         Input should be '<Kind>' [type=literal_error, input_value='Status', input_type=str]
E                           For further information visit https://errors.pydantic.dev/2.10/v/literal_error
```

**Expected behavior**
_A clear and concise description of what you expected to happen:_

It can raise error, or keep watching, but not this way

**Environment**
- Cloudcoil Version: 0.5.6
- Python Version: 3.12.4
- Kubernetes Version: 1.32.1.3

**Additional context**
Add any other context about the problem here.

--- a/cloudcoil/client/_api_client.py
+++ b/cloudcoil/client/_api_client.py
@@ -426,7 +426,6 @@ class APIClient(_BaseAPIClient[T]):
                             continue
                         event = json.loads(line)
                         type_ = event["type"]
-                        obj = self.kind.model_validate(event["object"])
 
                         if type_ == "ERROR":
                             if "status" in event["object"]:
@@ -444,6 +443,7 @@ class APIClient(_BaseAPIClient[T]):
                             logger.error("Watch error event received: %s", event)
                             raise WatchError(f"Watch error: {event}")
 
+                        obj = self.kind.model_validate(event["object"])
                         if obj.metadata and obj.metadata.resource_version:
                             curr_resource_version = obj.metadata.resource_version
                         yield type_, obj
@@ -855,7 +855,6 @@ class AsyncAPIClient(_BaseAPIClient[T]):
                             continue
                         event = json.loads(line)
                         type_ = event["type"]
-                        obj = self.kind.model_validate(event["object"])
 
                         if type_ == "ERROR":
                             if "status" in event["object"]:
@@ -873,6 +872,7 @@ class AsyncAPIClient(_BaseAPIClient[T]):
                             logger.error("Watch error event received: %s", event)
                             raise WatchError(f"Watch error: {event}")
 
+                        obj = self.kind.model_validate(event["object"])
                         if obj.metadata and obj.metadata.resource_version:
                             curr_resource_version = obj.metadata.resource_version
                         yield type_, obj

